compiler-core/src/language_server/completer.rs

```diff
@@ -535,6 +536,38 @@ where
         // Do not complete direct module values if the user has already started typing a module select.
         // e.x. when the user has typed mymodule.| we know local module values are no longer relevant
         if module_select.is_none() {
+            let cursor = self
+                .src_line_numbers
+                .byte_index(self.cursor_position.line, self.cursor_position.character);
+
+            // Function arguments and variables
+            // Find the function that the cursor is in
+            // and push the arguments and local variables
+            if let Some(fun) = self.module.ast.definitions.iter().find_map(|d| match d {
+                Definition::Function(f) if f.full_location().contains(cursor) => Some(f),
+                _ => None,
+            }) {
+                extract_args_completions(mod_name, insert_range, &fun.arguments, &mut completions);
+                completions.extend(extract_from_local_statements(
+                    mod_name,
+                    insert_range,
+                    cursor,
+                    &fun.body,
+                ));
+            }
+
+            for definition in &self.module.ast.definitions {
+                // Stop if we've passed the cursor position
+                if definition.location().start > cursor {
```

Comment:
Are definitions guaranteed to be ordered by location at this point? I think they are ordered by dependency here.